src/tests/system/tests/test_ipa.py

```diff
@@ -596,3 +596,41 @@ def test_ipa__idview_fails_to_apply_on_ipa_master(ipa: IPA):
     assert (
         "ID View cannot be applied to IPA master" in result.stdout
     ), "Did not get an error message when trying to apply ID view on server!"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__switch_user_with_smartcard_authentication(client: Client, ipa: IPA):
+    """
+    :title: Smart card authentication allows nested 'su' as IPA user
+    :setup:
+        1. Create IPA user with key pair and certificate
+        2. Copy certificate to client and initialize virtual smart card
+        3. Configure SSSD with smart card support and start SSSD services
+    :steps:
+        1. Execute 'su - ipacertuser1' using smart card authentication (PIN required)
+        2. From within the session, execute nested 'su - ipacertuser1 -c whoami'
```

Comment:
Hi,

the test looks good and is working as expected for me. But the description here is wrong or at least misleading. The first `su` is executed as `root` and be default this means that the switch to the given user happens without any authentication, this is because of `auth            sufficient      pam_rootok.so` in `/etc/pam.d/su`. The "PIN" prompt only happens during the second `su` because here is it run as ordinary user `ipacertuser1`.  Since the output of both commands is merge into a single string the assert passes as expected.

The missing authentication if the first `su` is run as root is the main reason to have this nested-`su`-pattern in the authentication tests.

bye,
Sumit